File: java/leetcode/flipping-an-image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] A) {
        if (A.length == 1 && A[0].length == 1) {
            A[0][0] = A[0][0] == 0 ? 1 : 0;
            return A;
        }
        int halfR = A.length > 0 ? (int)Math.floor(A[0].length / 2)  : 0;
        int temp = 0;
        for (int c = 0; c < A.length; c++) {
            for (int r = 0; r < halfR; r++) {
                temp = A[c][r] == 1 ? 0 : 1;
                A[c][r] = A[c][A[c].length - 1 - r] == 1 ? 0 : 1;
                A[c][A[c].length - 1 - r] = temp;
            }
            if (A[0].length % 2 == 1) {
                A[c][halfR] = A[c][halfR] == 1 ? 0 : 1;
            }
        }
        return A;
    }
}

File: python/leetcode/flipping-an-image.py
class Solution:
    def flipAndInvertImage(self, A):
        """
        :type A: List[List[int]]
        :rtype: List[List[int]]
        """
        if len(A) == 1 and len(A[0]) == 1:
            A[0][0] = 1 if A[0][0] == 0 else 0
            return A
        half_r = len(A[0]) // 2 if len(A) > 0 else 0
        temp = 0
        for c in range(len(A)):
            for r in range(half_r):
                temp = 0 if A[c][r] == 1 else 1
                A[c][r] = 0 if A[c][len(A[c]) - 1 - r] == 1 else 1
                A[c][len(A[c]) - 1 - r] = temp
            if len(A[0]) % 2 == 1:
                # invert middle value
                A[c][half_r] = 0 if A[c][half_r] == 1 else 1
        return A
